### PR TITLE
[RNMobile] Using deferredEventEmitter to wait for the Editor to be mounted before issuing replaceBlock signal

### PR DESCRIPTION
**Pull request by mzorz**
## Description
This PR aims at approaching the edition of Stories on mobile by using the `deferredEventEmitter` instead of passing a live callback, which would otherwise get lost if the GutenbergEditor gets killed when opening the Story composer (for Story block editing) on Android and the system decides to claim memory.

This approach seems to work, except for the following problem: the `clientId` of a block will not be the same anymore in between Editor startups, so we can't use the `blockId` as a means of identifying which Story block to replace. This is what happens on WPAndroid:
1. edit a block
2. the replaceBlock event gets queued for later, with the current blockId
3. the new Activity is created, the Gutenberg editor is mounted and the Post is loaded which means Gutenberg assigns new block ids to each block
4. the replaceBlock action is processed but the blockId simply doesn't exist anymore so the event is missed.

Given this, we're using some new bridge signals to let the native side know when a Story block has been loaded (after a Gutenberg fireup) passing its newly assigned clientId, and the native side can compare whether this block needs to be replaced with a new one or not by calculating a hash on its contents as seen in the [WPAndroid PR](https://github.com/wordpress-mobile/WordPress-Android/pull/13699).

In summary, this PR basically adds these changes:
- moves the replaceBlock to use deferredEventEmitter
- adds a new bridge method `mediaFilesBlockReplaceSync` that gets triggered every time a Story block is loaded

Unrelated changes
- removed unused callback parameter from `requestMediaFilesEditorLoad` 

## How has this been tested?

Use WPAndroid PR https://github.com/wordpress-mobile/WordPress-Android/pull/13699
Gutenberg mobile PR https://github.com/wordpress-mobile/gutenberg-mobile/pull/2949
Jetpack PR https://github.com/Automattic/jetpack/pull/18540
Related stories library PR https://github.com/Automattic/stories-android/pull/632

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/react-native-bridge/android/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/GutenbergBridgeJS2Parent.java [hunk 1]
```diff
@@ -166,14 +163,13 @@ void gutenbergDidRequestUnsupportedBlockFallback(ReplaceUnsupportedBlockCallback
 
     void onShowXpostSuggestions(Consumer<String> onResult);
 
-    void requestMediaFilesEditorLoad(ReplaceMediaFilesEditedBlockCallback replaceMediaFilesEditedBlockCallback,
-                                                     ReadableArray mediaFiles,
-                                                     String blockId
-    );
+    void requestMediaFilesEditorLoad(ReadableArray mediaFiles, String blockId);
```
Comment (mzorz) [inline packages/react-native-bridge/android/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/GutenbergBridgeJS2Parent.java:166]: unrelated to this PR, removed the ReplaceMediaFilesBlockCallback as it was not being used

**Review by jd-alexander — COMMENTED** [review]
@mzorz just an update here. I did a first pass on the code. I am going to do another review soon and run some tests so that I can verify the changes made here in connection with the other PRs.

**Review by jd-alexander — APPROVED** [review]
Thanks for these changes @mzorz Looks good to me! LGTM 🚢

File: packages/react-native-bridge/android/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/GutenbergBridgeJS2Parent.java (hunk 1 above)
Comment (jd-alexander) [reply packages/react-native-bridge/android/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/GutenbergBridgeJS2Parent.java:166]: makes sense! Thanks!

**Pull request merged**